### PR TITLE
Cherry-pick `uNetworking/uWebSockets#570fe762c82383b53765aeef3cafb65863c3abbc`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

This cherry-picks https://github.com/uNetworking/uWebSockets/commit/570fe762c82383b53765aeef3cafb65863c3abbc

Fixes https://github.com/oven-sh/bun/issues/3580

### How did you verify your code works?

CI

### CODE REVIEW

File: packages/bun-uws/src/HttpParser.h [hunk 1]
```diff
@@ -227,16 +227,15 @@ namespace uWS
             return unsignedIntegerValue;
         }
 
-        /* RFC 9110 16.3.1 Field Name Registry (TLDR; alnum + hyphen is allowed)
-         * [...] It MUST conform to the field-name syntax defined in Section 5.1,
-         * and it SHOULD be restricted to just letters, digits,
-         * and hyphen ('-') characters, with the first character being a letter. */
-        static inline bool isFieldNameByte(unsigned char x)
```
Comment (jdalton) [inline packages/bun-uws/src/HttpParser.h:234]: Comment above was helpful.

File: packages/bun-uws/src/HttpParser.h [hunk 2]
```diff
@@ -263,23 +262,19 @@ namespace uWS
...
+                while (isFieldNameByte(*(unsigned char *)p)) {
+                    *(p++) |= 0x20;
                 }
-                word |= 0x2020202020202020ull;
-                memcpy(p, &word, sizeof(uint64_t));
-            }
+                return (void *)p;
+            //}
+            //word |= 0x2020202020202020ull;
+            //memcpy(p, &word, sizeof(uint64_t));
```
Comment (jdalton) [inline packages/bun-uws/src/HttpParser.h:276]: Is the commented out code debug or likely to be reverted?

Comment (cirospaciari) [reply packages/bun-uws/src/HttpParser.h:276]: unlikely to be reverted but this is the official committed version:
https://github.com/uNetworking/uWebSockets/commit/570fe762c82383b53765aeef3cafb65863c3abbc

Comment (jdalton) [reply packages/bun-uws/src/HttpParser.h:276]: Ah I see. We are literally cherry-picking their patch.

**Pull request merged**